012_super_resolution/STORM.ipynb

```diff
@@ -0,0 +1,818 @@
OJ9222aHYtPklEoFSZLPoV7oGSCfEyNjguepUrYV/ffVY7LXUUu/WkTTGj12mwSXw4bZlYXQQ1LpKHrhe+ac7/Xg+QOtimEnGjpmfYRX2VsxrSzBHmLYJFnc3H6mF+2DI+h0+/DW6W7D8u++YBj/NEgMrdc2bUU7zuBIENvO9Cjhn88faFXlTOOIiY45Rj0nFIko1fneOt2NEx3ytoNhFtcfJQmYXVWIFdNKDbfPfxMKR9AYTaIfJ9waCmT6WNWdXzka7wEHBlUy9H4DQ10J34seQM+wH409nlj4nl3CtQuqsXpWueo34v3CRd3ReD+lMwwpnV5YeuGyemPF5pNdupXx9A5L+5GeJ6jDJgm50GL70xMytO3ccqo7zj6abiJC6J17UdQSnztm8NxpyXKyQxabRUcp7Zg/7A/h+QOt6v1J6r8Zk/M1bTnVbXivFTjtlp9VVkKlATl8T0wqL27f7OVRmEWfO8Iq3kDYcri3EfNqChOKwpUF5l6xWkiUIgiB+rJ8PHrbCkWk+tkbDfjY/+5U3nSlUqGGIAiCIMaKQa8c0iXO5fWmmjkmuXwAwTsnuqGggWFihp4IwMPnZ1fKb9H9oQiGfCG0DYzArklabJMkbD4lT/gZS676Hq/upoeVsLVk8trowT1WjNp4tG3QsBrfzsbeOGFKiy8YNnw7X57vRDDMEoYgMiQ+D3pCzNluj9IGQP2m3WxeVJybg1tXTE0qNNIWNZS7hnzY1din6xUmYpaHRS9czYr4aJaonBu3vmBYEQhEDxm989ul8QiMMIZejeeIWbNigrG88X4hHxz/WSjMMOwPCnmuYsj3kiSESFkTFbVoV42w2O+H/SE09RqEpwmeXk29Xrzd0INDLQNK35EAlOTlqO5BHqan9aCJ/RsrhFBflpxHRjpIpzWgF+qs9SI1Q8+DVNuf9O5Bjz+Efm8AgZD6+WHUJ8KCgKU3FmnvrWTOEb/OicysYJhpwliTuxJiG7XtTTakzojZVYWWvZGCJoUZRLqGfIYFB3i4sB6bjnfirdPq/IYRFhPFrQhTeqtYeV4m+0glUYogTPj2rYvx1L+uUlyP3//zbXh04ynl+3RVWCEIgiCIVNja0I1djX26XlFWDA2+Xpw3RnSybDnkhQHH2+OrR9kk2TuFV/XjYWbnej1K6XZOQ5ecY0T2lNLJKWWye1XJcQ0jwXBC49us1LwV+Dk301/ME9nqvwXniCKG9vvuYT8ijFmqTKZ3HkTDTkzIG9thdL3ooniujHKHpYroSSfu0wjTsFTNr+02Cae7jCuccYYthEK+crQDh6OeIqKAa8UT7UTHEM5rKjyahabyPsXt121nYkYmP/5+bwB+QVzQOy96BnwqSbc5Q74g2oXcY9qwWED2nonluGPY39yvfKet3ia22UjI1AuRm19TqIRliqiE+nFK1pwukrEvLOX80TmfI1GRuSc6fnD8Brl1IxqBML4h8j9XzK1EcW6Oecgy4oVTQBZfzCpjnuhwo1XwHBVFcSPRRkQMMdZW6rRyL1i5W5IR4ZO5/TYKqWXK8mMvYbqGfHD7giqvK7G/93sDceO0XN0WeM/yKbg46slq3Jb449Eeo9YzORVIlCIIE4pyc7Aqmvw1FI7gslnlinvqSCCMJQ+9gl9vO6cs/37neaWEKkEQBEGMB/3egGZCKaHT7VNNRE1FKcGVn0/srZaeFhkciX97HmFMtQ3+V/eQXwlB4qg9TuLDC8zyPpkZJO2DI9h9zvhtMiB7/6wVKp2lSjoSvuoaATrnkL8wU9axkJtH7yxpKwWKdA354vJHJfK6Gw02SYJoj2u7bTLFaLS/tert7tPJ22OGmK+He5WZoevxZtJt+HEcahmIC/fRXvLzfZ64NvHNx0KkYp+LeX0SoT17x9rdhtUutfvW7heQxRAg1j/FW4fbuLFk9fIfYm4rvn5Rbg7WXVSj6v9DGiM8URJrK6jCYMcyizp4qFjq+4gbOxNUgjTzjFWeDdFVjJ4l/BRXFrpw3cJq1fXQO5Z+bwB7z/crbeBsPW08HgHAwZYB5W8xWboV8pz2uM942Khfp8pkqnkM9TjaNhj3WaqifoHLgUumx8Ki3zjRhYMtA2gfHMGLh9oStjoSifVnrZe0Ft3wUAvPZvKUIogxwmG34YFbLsIHV9YDkB+Mn712DpbVlwAAznQP4+vPHsGx6AP6cMsgrvzP15UEpW0DI3jhYFva3yoSBEEQFzbhiFqSYYxhx9lelRFuJkrZJClm3GoMD6tvfbuH/bql48MR9eTUqmij5yllRp5GoElFGzLKB2UFvj+jpNupbEukR/Aq499fu6BKVV3YkUAsOto6qIg6Zp5lImKiY62hPxbYJEmVP0orQh3RGHZWLrMVDwrV+kleQofgJXCmO95INtteca7c58yOg9+LI8Gw4p3F0RrN3BNEvG/tNkmVIHo8E1Ew6CdqF5E0/wJCBVHNuk67zVBcVHmGRoCAoG6mQyx2Oey4PponTQ6FlXdYV6Jf3GC0WBUMrRya2Sr9ngB2nDVOaM67kpI/zHA94704DDxpeGVH8dolShgvYsXzUUSvEAX3sgsmEK3Nkqynwu5zfdh3fiBt2+vzBLCrsQ/hSOJ8kIFwTHivKJS9rowSk5v1nUuml2FWZXxxBiB570QSpQgiRcoKnLh3/XxcHFWqF9UVY+fX1uHKuXLpW6fDhtUzy1EVdV/dcbYX//7H/egblicMrx7twN2/3YvB6BtIvcomBEEQBGGFc4LXkV4YTZ/HWFAoyc9RjNtwtIoai9pzyVSl0ws5CUfDBDhW56lDvlBSwhI37pPFqKx8sihhVAYV6KwyOBLUDV9Rix3yvlwOO4rzYmFL2hxdWjrcPrwd9YqqKHQq8xU99ML4Yl4NYzdX0Rq2WqOzZ9i6Z8/RtsGUqiknMui0c7VkvXBE8ZB7bpiHIcY4pwn7M7IYxfOYTDiRGVpBzAo8n5VeVUUteuF72pxSHW4fXoomtjc7qs2nurD5ZKzSWbqi
```

Comment:
I think it would be useful to include a simple comparison with a poorer sampling result, to highlight the effect of the approach presented here.  How about a simple MH? Or a non-reduced or non-gradient employing method? How much work would it be to add such a simple comparison?

Reply:
I just want to remind that NUTS is only sampler available in CUQIpy-PyTorch. For experiments with non-gradient samplers, for example MH as you suggest, we need to re-implement the whole problem in CUQIpy and I'm afraid this would add confusion to a reader since the way to prepare a forward model and a custom distribution are a bit different in CUQIpy and CUQIpy-Pytorch (Generally in CUQIpy-PyTorch, we work with PyTorch tensors while in CUQIpy we work with numpy arrays).

Reply:
We decided at stand-up to drop the idea of performing any comparison in this notebook since:
- MH needs re-implementation of the problem in CUQIpy;
- NUTS with the original high-dimensional likelihood is very computationally expensive (after an attempt running of 3 days);
- The current notebook is already a complete story.

At the end of the notebook, I added the following comment:
> While a comparison of applying a reduced likelihood and the original likelihood with NUTS would be insightful, it is not performed in this notebook due to the computational demands of UQ with the high-dimensional original likelihood.